homeassistant/components/compit/climate.py

```diff
@@ -0,0 +1,284 @@
+                )
+            )
+            is not None
+            if (device_definition.device_class == CLIMATE_DEVICE_CLASS)
+        ]
+    )
+
+
+class CompitClimate(CoordinatorEntity[CompitDataUpdateCoordinator], ClimateEntity):
+    """Representation of a Compit climate device."""
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.COOL, HVACMode.OFF]
+    _attr_name = None
+    _attr_has_entity_name = True
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.FAN_MODE
+        | ClimateEntityFeature.PRESET_MODE
+    )
+
+    def __init__(
+        self,
+        coordinator: CompitDataUpdateCoordinator,
+        device: Device,
+        parameters: list[Parameter],
+        device_name: str,
+    ) -> None:
+        """Initialize the climate device."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{device.label}_{device.id}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, str(device.id))},
+            name=device.label,
+            manufacturer=MANUFACTURER_NAME,
+            model=device_name,
+        )
+
+        parametersDict = {
+            parameter.parameter_code: parameter for parameter in parameters
+        }
+        self.device_id = device.id
+        self.available_presets: Parameter | None = parametersDict.get(PARAM_PRESET_MODE)
+        self.available_fan_modes: Parameter | None = parametersDict.get(PARAM_FAN_MODE)
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        value = self.get_parameter_value(PARAM_CURRENT_TEMPERATURE)
+        if value is None:
+            _LOGGER.warning("The current_temperature is None")
+            return None
+        return float(value.value)
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        value = self.get_parameter_value(PARAM_TARGET_TEMPERATURE)
+        if value is None:
+            _LOGGER.warning("The target_temperature is None")
+            return None
+        return float(value.value)
+
+    @cached_property
+    def preset_modes(self) -> list[str] | None:
+        """Return the available preset modes."""
+        if self.available_presets is None or self.available_presets.details is None:
+            return []
+        return [
+            item.description
+            for item in self.available_presets.details
+            if item is not None
+        ]
+
+    @cached_property
+    def fan_modes(self) -> list[str] | None:
+        """Return the available fan modes."""
+        if self.available_fan_modes is None or self.available_fan_modes.details is None:
+            return []
+        return [
+            item.description
+            for item in self.available_fan_modes.details
+            if item is not None
+        ]
```

Comment:
what are the possible fan modes? We have a couple we use as common keys, and they already have icons and strings, so maybe we can map to those